cg/services/analysis_starter/configurator/implementations/balsamic.py

```diff
@@ -0,0 +1,259 @@
+            swegen_sv=self.swegen_sv,
+            tumor_sample_name=self._get_tumor_sample_id(case),
+        )
+
+    @staticmethod
+    def _all_samples_are_wgs(case: Case) -> bool:
+        """Check if all samples in the case are WGS."""
+        return all(
+            sample.application_version.application.prep_category
+            == SeqLibraryPrepCategory.WHOLE_GENOME_SEQUENCING
+            for sample in case.samples
+        )
+
+    @staticmethod
+    def _all_samples_are_exome(case: Case) -> bool:
+        """Check if all samples in the case are exome."""
+        return all(
+            sample.application_version.application.prep_category
+            == SeqLibraryPrepCategory.WHOLE_EXOME_SEQUENCING
+            for sample in case.samples
+        )
+
+    @staticmethod
+    def _get_patient_sex(case) -> SexOptions:
+        sample_sex: set[SexOptions] = {sample.sex for sample in case.samples}
+        if len(sample_sex) == 1:
+            return sample_sex.pop()
+        else:
+            raise BalsamicInconsistentSexError(
+                f"Case {case.internal_id} contains samples of differing sex"
+            )
+
+    @staticmethod
+    def _get_normal_sample_id(case) -> str | None:
+        for sample in case.samples:
+            if not sample.is_tumour:
+                return sample.internal_id
+
+    @staticmethod
+    def _get_tumor_sample_id(case) -> str:
+        for sample in case.samples:
+            if sample.is_tumour:
+                return sample.internal_id
+        raise BalsamicMissingTumorError(f"Case {case.internal_id} does not contain a tumor sample")
+
+    def _resolve_bed_file(self, case, **flags) -> Path:
+        bed_name = flags.get("panel_bed") or self._get_bed_name_from_lims(case)
+        if db_bed := self.store.get_bed_version_by_short_name(bed_name):
+            return Path(self.bed_directory, db_bed.filename)
+        raise BedFileNotFoundError(f"No Bed file found for with provided name {bed_name}.")
+
+    def _get_bed_name_from_lims(self, case: Case) -> str:
+        """Get the bed name from LIMS. Assumes that all samples in the case have the same panel."""
+        first_sample = case.samples[0]
+        if lims_bed := self.lims_api.capture_kit(lims_id=first_sample.internal_id):
+            return lims_bed
+        else:
+            raise BedFileNotFoundError(
+                f"No bed file found in LIMS for sample {first_sample.internal_id} in for case {case.internal_id}."
+            )
+
+    def _get_pon(self, bed_file: Path) -> Path:
+        """Finds the corresponding PON file for the given bed file.
+        These are versioned and named like: <bed_file_name>_hg19_design_CNVkit_PON_reference_v<version>.cnn
+        This method returns the latest version of the PON file matching the bed name.
+        """
+        identifier = bed_file.stem
+        pattern = re.compile(rf"{re.escape(identifier)}.*_v(\d+)\.cnn$")
+        candidates = []
+
```

Comment:
type hints here would help

Reply:
Added in 2ec3d5ef283e9d74c5cf5bf1441426550c6e1e22